### PR TITLE
feat: improve auth example

### PR DESCRIPTION
**Pull request by juanpicado**
Add auth example with notes.

Requires: https://github.com/verdaccio/monorepo/pull/305

### CODE REVIEW

File: generators/app/templates/typescript/auth/src/index.ts [hunk 1]
```diff
@@ -26,29 +29,59 @@ export default class AuthCustomPlugin implements IPluginAuth<CustomConfig> {
...
+    /**
+     * This code is just an example for demostration purpose
+     */
+    if (user.name === this.foo && pkg.access.includes[user.name]) {
+      this.logger.debug({name: user.name}, 'your package has been granted for @{name}');
+      cb(null, ['group-foo', 'another-group-foo'])
+    } else {
+      this.logger.error({name: user.name}, '@{name} is not allowed to access this package');
+      cb('error, try again', false);
+    }
```
Comment (anikethsaha) [inline generators/app/templates/typescript/auth/src/index.ts]: Will it make sense to add the example code inside the comment?

```js
/**
 * This code is just an example for demostration purpose
 * if (user.name === this.foo && pkg.access.includes[user.name]) {
 *   this.logger.debug(
 *     { name: user.name },
 *     "your package has been granted for @{name}"
 *   );
 * cb(null, ["group-foo", "another-group-foo"]);
 *   } else {
 *    this.logger.error(
 *      { name: user.name },
 *      "@{name} is not allowed to access this package"
 *  );
 *  cb("error, try again", false);
 * }
 */
```

Comment (juanpicado) [reply generators/app/templates/typescript/auth/src/index.ts]: Yes but, might be in the header, why I wan to achieve is provide the basic compile code so when run yarn build the project compiles. I can move the full example to the header as comment and leave below the most basic response, it is ok ?

Comment (anikethsaha) [reply generators/app/templates/typescript/auth/src/index.ts]: I do see the point of kickstarting with a compile friendly code. I am fine with either way.
both seem fine to me 👍

Comment (juanpicado) [reply generators/app/templates/typescript/auth/src/index.ts]: better now?

Comment (anikethsaha) [reply generators/app/templates/typescript/auth/src/index.ts]: Yeah 👍

**Pull request merged**